cores/esp8266/umm_malloc/umm_malloc.cpp

```diff
@@ -498,9 +498,30 @@
 #include "umm_malloc.h"
 
 #include "umm_malloc_cfg.h"   /* user-dependent */
+#if defined(DEBUG_ESP_PORT)
+#include <assert.h>
+#endif
 
 extern "C" {
 
+#ifdef UMM_CRITICAL_PERIOD_ANALYZE
```

Comment:
There is currently no assurance that these performance functions will be accepted upstream. We can propose them anyways, but it could take a while for the code to be reviewed and tested.
Instead, I suggest moving into a umm performance.h|cpp files, including the .h in the config file, and adjusting the build as/if necessary for the extra .cpp.
The above is possible thanks to the use of the DECL, ENTRY and EXIT macro hooks.

Reply:
Okay